yt/frontends/fits/data_structures.py

```diff
@@ -109,25 +110,19 @@ def _guess_name_from_units(self, units):
                 return v
         return None
 
-    def _determine_image_units(self, header, known_units):
+    def _determine_image_units(self, header):
         try:
-            field_units = header["bunit"].lower().strip(" ").replace(" ", "")
-            # FITS units always return upper-case, so we need to get
-            # the right case by comparing against known units. This
-            # only really works for common units.
-            units = set(re.split(regex_pattern, field_units))
-            if '' in units: 
-                units.remove('')
-            n = int(0)
-            for unit in units:
-                if unit in known_units:
-                    field_units = field_units.replace(unit, known_units[unit])
-                    n += 1
-            if n != len(units) or n == 0:
-                field_units = "dimensionless"
-            if field_units[0] == "/":
-                field_units = "1%s" % field_units
-            return field_units
+            try:
```

Comment:
Is there any way we could flatten this?

Reply:
If possible, flattening this would make it a bit more readable!